src/Typeclasses/Classes/Bounded.elm

```diff
@@ -0,0 +1,123 @@
+module Typeclasses.Classes.Bounded exposing
+    ( Bounded
+    , interval
+    , int, char, bool, order, unit
+    , tuple2, tuple3
+    )
+
+{-| Bounded typeclass definition and its instances for basic types.
+
+
+# Definition
+
+@docs Bounded
+
+
+# Construction utilities
+
+@docs interval
+
+
+# Instances
+
+@docs int, char, bool, order, unit
+
+
+# Composites
+
+@docs tuple2, tuple3
+
+-}
+
+
+{-| Explicit typeclass which implements a bounded type `a`.
+-}
+type alias Bounded a =
+    { minBound : a
+    , maxBound : a
+    }
+
+
+
+-- * Constructors
+-------------------------
+
+
+{-| Construct an instance from the interval `(minBound, maxBound)`.
+-}
+interval : ( a, a ) -> Bounded a
```

Comment:
Why do you use Tuple here?

Reply:
It seemed more natural for numeric types, but I can see it being tedious to work with otherwise.

I could change that to `interval : a -> a -> Bounds a`, but it would just be an alias for the `Bounds` record constructor.

Reply:
Right. Seems like this function is altogether redundant. It's just an alias to `uncurry Bounds`. See https://package.elm-lang.org/packages/TSFoster/elm-tuple-extra/latest/Tuple2#uncurry

Reply:
I pushed before I was aware of your comment. I find `bounds: a -> a -> Bounds a` to be more readable than before.

Would it be best to remove it altogether or is that unnecessary?